Correct type of bdd_cuddt::indext

romainbrenguier opened the pull request:
This was causing errors when compiling with clang -Werror -Wsign-conversion

<!---
Thank you for your contribution. Please make sure your pull request fulfils all of the below requirements. If you cannot currently tick all the boxes, but would still like to create a PR, then add the label "work in progress" and assign the PR to yourself.
--->

- [x] Each commit message has a non-empty body, explaining why the change was made.
- [na] Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- [na] The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [ ] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- [na] My commit message includes data points confirming performance improvements (if claimed).
- [x] My PR is restricted to a single feature or bugfix.
- [na] White-space or formatting changes outside the feature-related changed lines are in commits of their own.

<!---
See, e.g., https://chris.beams.io/posts/git-commit/ for general guidelines on commit messages.

If you have created commits mixing multiple features and/or unrelated white-space changes, use a sequence involving git reset and git add -p to fix this.
--->

romainbrenguier committed:
Correct type of bdd_cuddt::indext

This was causing errors when compiling with clang -Werror -Wsign-conversion

allredj reviewed (commented):
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: 871c551a6d76955982ca8aa7ab8e0e32cadc167f).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/103720888

romainbrenguier committed:
Correct argument type of cudd.bddVar

The type of cudd.bddVar differs from cuddNodeReadIndex, so we need a
conversion between the two.

tautschnig commented on src/solvers/bdd/bdd_cudd.h:
```diff
@@ -152,7 +152,7 @@ class bdd_managert
   bddt bdd_variable(bdd_nodet::indext index)
   {
-    return bddt(cudd.bddVar(index));
+    return bddt(cudd.bddVar(narrow_cast<int>(index)));
```
You need `#include <util/narrow.h>` to get this to build.

allredj reviewed (commented):
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: 52ba6c921e9e615c5cab0ababf201bcc01d2ea6a).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/103862848
